### ISSUE TITLE
Odd field sampling

### ISSUE THREAD

**Issue by Jacketless** [OceanPARCELS/parcels#74]
I've been getting some strange results when sampling from none U/V forcing fields, and I'm not sure where they were coming from.

I have a `Field` object, with data that I have checked is > 0 in all cases. Occasionally however, when a kernel samples this field for continuous [time, lon, lat] values I'm given a negative number by the interpolator. This can still occur when I remove the RK4 temporal interpolation and use discrete time.

Is there a way that the `RectBivariateSpline' function can be somehow returning me negative numbers? Perhaps it is interpolating with null cell values present in some points on the field...

**Comment by mlange05** [comment]
This type of problem is virtually impossible to track down without a [minimal working example](https://en.wikipedia.org/wiki/Minimal_Working_Example). Can you please provide a (down-sampled) grid file and a gist that demonstrates this potential bug? Also, have you run this with JIT by any chance?

**Comment by Jacketless** [comment]
Yes sorry I'm aware that bug tracking is difficult without this. I had started writing minimal example, but ran out of time and thought it better to at least send an email off. I will try and put something together over the weekend with a downscaled field.

**Comment by Jacketless** [comment]
There is now a minimal working example of this as part of PR #77 . It's possible that there is a casting issue going on, although some of the returned values from grid sampling are well below the grid minimum.

I'm aware this example is not something we necessarily want to merge, but I did it this way in case it later becomes part of an internal test.

**Comment by Jacketless** [comment]
Having spent a little more time looking into this, I've narrowed the strange field sampling down to be a result of using Scipy's `RectBivariateSpline`. When there is a strong gradient and vertex values are close to zero, the spline can return negative values when interpolating between all positive values.

I've got a temporary fix going using linear 2d interpolation `interp2d` from Scipy, but this raises the issue of us perhaps wanting to use linear interpolation when a spline evaluation returns values outside a range specified by the user (although this would be slow). Alternatively, users could just specify interpolation on each `Field` object?
